docs/.eleventy.js

```diff
@@ -19,10 +19,10 @@ module.exports = function(eleventyConfig) {
      * The site is loaded from /docs on eslint.org and so we need to adjust
      * the path prefix so URLs are evaluated correctly.
      *
-     * The path prefix is turned off for deploy previews so we can properly
+     * The path prefix is turned off for `npm start` and deploy previews so we can properly
      * see changes before deployed.
      */
-    const pathPrefix = process.env.CONTEXT === "deploy-preview" ? "" : "/docs";
+    const pathPrefix = ["local-serve", "deploy-preview"].includes(process.env.CONTEXT) ? "" : "/docs";
```

Comment:
I couldn't make it work (see the third point in "What changes did you make" in the original post). I'll see if the browser-sync config override can be done differently.

Reply:
Huh. I haven’t had any trouble since I first added pathPrefix. Eleventy generates an index.html file at the root that redirects to /docs and then everything works perfectly for me. I’m using the same setup on the playground.

Reply:
Eleventy's built-in server returns the content of `foo/index.html` when we request `foo/`, but it doesn't return the content of `foo/bar.html` when we request `foo/bar`, which is what we'll have after this change. It needs some additional configuration.

Without pathPrefix, it works well when I add `serveStaticOptions: { extensions: [ 'html' ] }` to the config that eleventy passes into the underlying server.

```js
eleventyConfig.setBrowserSyncConfig({
    server: {
        baseDir: "_site",
        serveStaticOptions: {
            extensions: ["html"]
        }
    }
});
```

With pathPrefix, for some reason it doesn't work.

```js
eleventyConfig.setBrowserSyncConfig({
    server: {
        baseDir: "_site/_eleventy_redirect",
        routes: {
            "/docs": "_site"
        },
        serveStaticOptions: {
            extensions: ["html"]
        }
    }
});
```

```
Cannot GET /docs/user-guide/configuring/configuration-files
```